### PR TITLE
BME68X gas sensor driver using BSEC

### PR DESCRIPTION
**Pull request by maxd-nordic**
This PR adds a Zephyr Sensors compatible driver for the gas sensor used in the Thingy:91 and Thingy:53 to NCS.
It calculates air quality values using Bosch's proprietary BSEC library and hides this behind a standard driver interface.

The BSEC library still has to be downloaded manually due to licensing issues. At least, we can refer to a specific tag in their new GitHub repository for that purpose. The old version of the library currently used in the asset tracker is impossible to obtain for a while now.

This supercedes https://github.com/nrfconnect/sdk-nrf/pull/9968.
Adresses CIA-866.

### CODE REVIEW

File: samples/sensor/bme68x/boards/thingy53_nrf5340_cpuapp.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Segger RTT
+CONFIG_USE_SEGGER_RTT=y
```
Comment (gregersrygg) [inline samples/sensor/bme68x/boards/thingy53_nrf5340_cpuapp.conf:8]: Why is RTT default for thingy:53 instead of UART?

File: samples/sensor/bme68x/prj.conf [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+CONFIG_SENSOR=y
+CONFIG_SENSOR_INFO=y
+
+CONFIG_BME680=n
```
Comment (gregersrygg) [inline samples/sensor/bme68x/prj.conf:10]: Why is this necessary? Is it enabled because of the device tree entry?

File: samples/sensor/bme68x/README.rst
Comment (gregersrygg) [inline samples/sensor/bme68x/README.rst:1]: I've pointed out the name of the driver before, but I would like the sample folder name to align with the new driver name. Now there's inconsistency between bme68x, bme68x_ncs and bsec.

File: samples/sensor/bme68x/README.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
+.. _bme68x:
+
+Gas Sensor BME688
```
Comment (gregersrygg) [inline samples/sensor/bme68x/README.rst:3]: BME68X?

File: samples/sensor/bme68x/README.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+###############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This sample application sets up the BME68X gas sensor with the Bosch Sensor Environmental Cluster (BSEC) library.
```
Comment (gregersrygg) [inline samples/sensor/bme68x/README.rst:10]: BSEC2?

File: samples/sensor/bme68x/README.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+Requirements
+************
+
+Using this driver requires manually downloading the BSEC library. See :ref:`lib_bme68x` for more details.
```
Comment (gregersrygg) [inline samples/sensor/bme68x/README.rst:15]: BSEC2?

File: samples/sensor/bme68x/README.rst [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+The sample supports the following devices:
+
+.. table-from-sample-yaml::
+
+Building and running
+********************
+
+This project outputs sensor data to the console.
+It requires a BME68X sensor.
+Board overlays for Thingy:91 and Thingy:53 are included.
```
Comment (gregersrygg) [inline samples/sensor/bme68x/README.rst:25]: I only see board overlay files for Thingy:53, but I don't think this sentence is needed. @divipillai ?

File: samples/sensor/bme68x/README.rst [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+The following output is displayed in the terminal:
+
+.. code-block:: console
+
+    *** Booting Zephyr OS build v3.2.99-ncs1-1531-gaf18f6b63608 ***
+    [00:00:01.285,339] <inf> app: App started
+    [00:00:07.287,658] <inf> app: temp: 28.240385; press: 100043.570312; humidity: 19.981348; iaq: 50
+    [00:00:10.301,391] <inf> app: temp: 28.220613; press: 100039.585937; humidity: 19.983814; iaq: 50
+    [00:00:13.315,124] <inf> app: temp: 28.188013; press: 100040.007812; humidity: 20.015941; iaq: 50
```
Comment (gregersrygg) [inline samples/sensor/bme68x/README.rst:42]: Should it mention in the sample that the iaq value takes ~24 hours to auto-calibrate?

File: samples/sensor/bme68x/sample.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+tests:
+  sample.sensor.bme68x:
+    depends_on: i2c
+    harness: sensor
+    integration_platforms:
+      - thingy91_nrf9160
+      - thingy91_nrf9160_ns
+      - thingy53_nrf5340_cpuapp
+      - thingy53_nrf5340_cpuapp_ns
+    platform_allow: >-
```
Comment (gregersrygg) [inline samples/sensor/bme68x/sample.yaml:12]: I don't know the sample.yaml format very well, but I think it's possible to add simple tests that look for console output using only yaml.

**Review by gregersrygg — COMMENTED** [review]
Sorry for taking so long to review. I had quite a few comments/questions, but I think that is to be expected for a PR of this size. Overall it is looking good!

Have you looked into unit-testing the driver/sensor? I found out it's possible to simulate a sensor in Zephyr while reviewing. A challenge could be that we don't have the bsec library available very easily because of the licensing issues. Do we have any tests in CI?

Feel free to reach out if you disagree or have any questions regarding my comments :)

**Comment by gregersrygg** [comment]
I think you also should add yourself to [CODEOWNERS](https://github.com/nrfconnect/sdk-nrf/blob/main/CODEOWNERS) for the driver and sample @maxd-nordic ?

File: samples/sensor/bme68x/README.rst (hunk 6 above)
Comment (divipillai) [reply samples/sensor/bme68x/README.rst:25]: If this is included to show the boards that support the sample, we could remove this sentence since it is mentioned in the requirement section, or is there any reason to mention the board overlays here?

File: samples/sensor/bme68x/README.rst [hunk 9]
```diff
@@ -0,0 +1,48 @@
+.. _bme68x:
+
```
Comment (divipillai) [inline samples/sensor/bme68x/README.rst:2]: The sample is now placed in the "Other samples" section. Since now, we have two sensor samples:  BH1749: Ambient Light Sensor IC and this sample. We could have a new sample category : Sensor samples in docs.

**Review by divipillai — COMMENTED** [review]
Shouldn't we add a change log entry for this PR?

File: drivers/sensor/bme68x/bme68x_ncs.c
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:1]: I'm open for suggestions. The current name makes sense to me, especially considering there is already a `bme68x.c`.

[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:1]: I think that would be very confusing, because `bme68x_ncs` looks like it's part of the official Bosch libraries. Maybe we could call it `bme_68x_zephyr` if we don't like to call it NCS.

File: drivers/sensor/bme68x/Kconfig [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config BSEC
+	bool "Use Bosch BSEC library"
+	depends on !BME680
+	depends on SETTINGS && !SETTINGS_NONE
+	depends on I2C
+	help
+	  Enable the use of Bosch BSEC library.
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/Kconfig:13]: It is BME68x + BSEC + lot's of glue to make it look like a Zephyr device.

File: drivers/sensor/bme68x/bme68x_ncs.h [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+#include <zephyr/kernel.h>
+#include "bsec_interface.h"
+#include "bme68x.h"
+#include <drivers/bme68x_ncs.h>
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_BME68X_NCS
+#define ZEPHYR_DRIVERS_SENSOR_BME68X_NCS
+
+#if defined(CONFIG_BSEC_SAMPLE_RATE_ULTRA_LOW_POWER)
+#define BSEC_SAMPLE_RATE BSEC_SAMPLE_RATE_ULP
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.h:18]: I'm not convinced that the CMakeLists file is a better place for this. We would also need 6 zephyr_library_compile_definitions_ifdef calls.

File: drivers/sensor/bme68x/bme68x_ncs.c
[the remark this answers is not in the document]
Comment (gregersrygg) [reply drivers/sensor/bme68x/bme68x_ncs.c:1]: I think `bme68x_zephyr` might be confusing with the bme68x driver in Zephyr upstream. I don't think it matters that much who wrote the driver, but what the driver supports. This driver is a combination of the bme68x sensor and the BSEC2 library, so I think `bme68x_bsec` or something similar would communicate to users what this driver does.

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 12]
```diff
@@ -0,0 +1,510 @@
...
+
+/* I2C bus write forwarder for bme68x driver */
+static int8_t bus_write(uint8_t reg_addr, const uint8_t *reg_data_ptr, uint32_t len, void *intf_ptr)
+{
+	uint8_t buf[len + 1];
+
+	buf[0] = reg_addr;
+	memcpy(&buf[1], reg_data_ptr, len);
+
+	return i2c_write_dt(&bme68x_i2c_spec, buf, len + 1);
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:125]: I'll use ARRAY_SIZE

File: drivers/sensor/bme68x/Kconfig [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+config BSEC_THREAD_STACK_SIZE
+	int "BSEC thread stack size"
+	default 4096
+
+config BSEC_EXPECTED_AMBIENT_TEMP
+	int "Expected ambient temperature in C"
+	default 25
+
+config BSEC_TEMPERATURE_OFFSET
+	int "BSEC temperature offset in degree celsius * 100"
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/Kconfig:43]: To be honest, introducing two significant digits is already a stretch with the accuracy of temperature measurements on this device. This is just transferred over from other implementations. I've never heard millidegrees before, so it's just as odd as centidegrees or similar.

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 14]
```diff
@@ -0,0 +1,510 @@
...
+		inputs[i].sensor_id = BSEC_INPUT_HEATSOURCE;
+		inputs[i].signal = temp_offset;
+		inputs[i].time_stamp = timestamp_ns;
+		LOG_DBG("Temp offset: %.2f", inputs[i].signal);
+		i++;
+		inputs[i].sensor_id = BSEC_INPUT_TEMPERATURE;
+		inputs[i].signal = data->temperature;
+
+		if (IS_ENABLED(BME68X_DO_NOT_USE_FPU)) {
+			inputs[i].signal /= 100.0f;
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:192]: I'll add a comment, but this is not documented in BME68x/BSEC either, just used in this way.
Also replacing this with a define does not make sense to me in this case.

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 15]
```diff
@@ -0,0 +1,510 @@
...
+		inputs[i].time_stamp = timestamp_ns;
+		LOG_DBG("Temp: %.2f", inputs[i].signal);
+		i++;
+	}
+	if (BSEC_INPUT_PRESENT(sensor_settings, BSEC_INPUT_HUMIDITY)) {
+		inputs[i].sensor_id = BSEC_INPUT_HUMIDITY;
+		inputs[i].signal =  data->humidity;
+
+		if (IS_ENABLED(BME68X_DO_NOT_USE_FPU)) {
+			inputs[i].signal /= 1000.0f;
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:204]: see above

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 16]
```diff
@@ -0,0 +1,510 @@
...
+		}
+
+		inputs[i].time_stamp = timestamp_ns;
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:207]: with different values for i. I'll add a comment.

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 17]
```diff
@@ -0,0 +1,510 @@
...
+	heater_config.heatr_dur_prof = sensor_settings.heater_duration_profile;
+	heater_config.profile_len = sensor_settings.heater_profile_len;
+	heater_config.shared_heatr_dur = 0;
+
+	switch (sensor_settings.op_mode) {
+	case BME68X_PARALLEL_MODE:
+		heater_config.shared_heatr_dur =
+			BSEC_TOTAL_HEAT_DUR -
+			(bme68x_get_meas_dur(sensor_settings.op_mode, &config, &data->dev)
+			/ INT64_C(1000));
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:260]: I don't know what any of the obfuscated calculations in bme68x do. This is from `bme68xLibrary.cpp:232`.

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 18]
```diff
@@ -0,0 +1,510 @@
...
+
+	heater_config.enable = BME68X_ENABLE;
+	heater_config.heatr_temp = sensor_settings.heater_temperature;
+	heater_config.heatr_dur = sensor_settings.heater_duration;
+	heater_config.heatr_temp_prof = sensor_settings.heater_temperature_profile;
+	heater_config.heatr_dur_prof = sensor_settings.heater_duration_profile;
+	heater_config.profile_len = sensor_settings.heater_profile_len;
+	heater_config.shared_heatr_dur = 0;
+
+	switch (sensor_settings.op_mode) {
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:255]: The order is important and cannot be changed. This is a procedure that talks to the device.
The first block is only used for one mode, the second for two, and the third for all modes. We could implement this using ifs, but that would be quite verbose.

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 19]
```diff
@@ -0,0 +1,510 @@
...
+	return 0;
+}
+
+/* Manage all recurrings tasks for the sensor:
+ * - update device settings according to BSEC
+ * - fetch measurement values
+ * - update BSEC state
+ * - periodically save BSEC state to flash
+ */
+static void bsec_thread_fn(const struct device *dev)
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:302]: Why would we want that?

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 20]
```diff
@@ -0,0 +1,510 @@
...
+	bsec_bme_settings_t sensor_settings = {0};
+	bsec_input_t inputs[BSEC_MAX_PHYSICAL_SENSOR] = {0};
+	bsec_output_t outputs[ARRAY_SIZE(bsec_requested_virtual_sensors)] = {0};
+	uint8_t n_fields = 0;
+	uint8_t n_outputs = 0;
+	uint8_t n_inputs = 0;
+	struct bme68x_ncs_data *data = dev->data;
+
+	while (true) {
+		uint64_t timestamp_ns = k_ticks_to_ns_floor64(k_uptime_ticks());
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:315]: The library explicitly uses nanoseconds and I wanted to honor that.

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 21]
```diff
@@ -0,0 +1,510 @@
...
+		if (sensor_settings.trigger_measurement &&
+		    sensor_settings.op_mode != BME68X_SLEEP_MODE) {
+			ret = bme68x_get_data(sensor_settings.op_mode,
+					      sensor_data, &n_fields, &data->dev);
+			if (ret) {
+				LOG_DBG("bme68x_get_data err: %d", ret);
+				continue;
+			}
+
+			for (size_t i = 0; i < n_fields; ++i) {
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/bme68x/bme68x_ncs.c:343]: It's perfectly fine to use ++i here. There is no point in using the post-decrement operator in a simple for-loop. If you feel that this should be a style requirement, add it to the linter checks and change the 153 occurrences in sdk-nrf.

File: doc/nrf/drivers/bme68x.rst [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+   :local:
+   :depth: 2
+
+You can use the BME68X driver to get gas measurements and run the Bosch Sensor Environmental Cluster (BSEC) library
+for IAQ values.
+
+.. note::
+   Using the BSEC library requires accepting a separate license agreement.
+   For details, see `BSEC2`_.
+   Since distribution of BSEC is not allowed, you need to manually check out the repository to the path specified by :ref:`CONFIG_BSEC_PATH <CONFIG_BSEC_PATH>` option. The currently tested version tag is "v1.5.2400".
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply doc/nrf/drivers/bme68x.rst:16]: Let's use BSEC instead of BSEC2.

File: samples/sensor/bme68x/boards/thingy53_nrf5340_cpuapp.conf (hunk 1 above)
Comment (maxd-nordic) [reply samples/sensor/bme68x/boards/thingy53_nrf5340_cpuapp.conf:8]: see above

File: samples/sensor/bme68x/prj.conf (hunk 2 above)
Comment (maxd-nordic) [reply samples/sensor/bme68x/prj.conf:10]: It's enabled by default in it's KConfig definition.

File: samples/sensor/bme68x/sample.yaml (hunk 8 above)
Comment (maxd-nordic) [reply samples/sensor/bme68x/sample.yaml:12]: do you have an example for this?

File: drivers/sensor/bme68x/bme68x_ncs.c (hunk 14 above)
[the remark this answers is not in the document]
Comment (gregersrygg) [reply drivers/sensor/bme68x/bme68x_ncs.c:190]: Ok. It seems like just a number they chose to multiply/divide by when there's no FPU. How about `NO_FPU_DIVIDER`?

File: drivers/sensor/bme68x/bme68x_ncs.c (hunk 15 above)
[the remark this answers is not in the document]
Comment (gregersrygg) [reply drivers/sensor/bme68x/bme68x_ncs.c:203]: Ok. It seems like just a number they chose to multiply/divide by when there's no FPU. How about `NO_FPU_HUMID_DIVIDER`?

File: drivers/sensor/bme68x/bme68x_ncs.c [hunk 23]
```diff
@@ -0,0 +1,510 @@
...
+static size_t sensor_data_to_bsec_inputs(bsec_bme_settings_t sensor_settings,
+					 const struct bme68x_data *data,
+					 bsec_input_t *inputs, uint64_t timestamp_ns)
+{
+	size_t i = 0;
+
+	if (BSEC_INPUT_PRESENT(sensor_settings, BSEC_INPUT_TEMPERATURE)) {
+		inputs[i].sensor_id = BSEC_INPUT_HEATSOURCE;
+		inputs[i].signal = temp_offset;
+		inputs[i].time_stamp = timestamp_ns;
```
[the remark this answers is not in the document]
Comment (gregersrygg) [reply drivers/sensor/bme68x/bme68x_ncs.c:180]: For every sensor we're setting the same 3 properties, but almost every block is a bit different. So it increases the cognitive load to see what the difference is. When extracting the same logic into a separate function that makes it much easier for the reader to see where the differences between each sensor is by just glancing at it.

The edge cases can still be handled on the outside of the extracted function:

```c++
float temp = data->temperature;
if (IS_ENABLED(BME68X_DO_NOT_USE_FPU)) {
	temp /= 100.0f;
}
set_input_data(inputs[i], BSEC_INPUT_TEMPERATURE, temp, time_stamp);
```

File: applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c [hunk 24]
```diff
@@ -11,8 +11,8 @@
 #include <stdlib.h>
 #include <math.h>
 
-#if defined(CONFIG_EXTERNAL_SENSORS_BME680_BSEC)
-#include "ext_sensors_bsec.h"
+#if defined(CONFIG_BSEC)
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c]: I currently don't see a good way to do this.

File: samples/sensor/bme68x/README.rst (hunk 9 above)
Comment (maxd-nordic) [reply samples/sensor/bme68x/README.rst:2]: Sure! How do we do this?

File: samples/sensor/bme68x/README.rst [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+********************
+
+This project outputs sensor data to the console.
+It requires a BME68X sensor.
+Board overlays for Thingy:91 and Thingy:53 are included.
+
+.. |sample path| replace:: :file:`samples/sensor/bme68x`
+
+.. include:: /includes/build_and_run.txt
+
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply samples/sensor/bme68x/README.rst:30]: This should just work after flashing, printing the results in the console. Should I add that section anyway?

[the remark this answers is not in the document]
Comment (divipillai) [reply samples/sensor/bme68x/README.rst:30]: Yeah, That sound good. This is what we follow in other sample documentation also: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/caf_sensor_manager/README.html#testing

[the remark this answers is not in the document]
Comment (maxd-nordic) [reply samples/sensor/bme68x/README.rst:30]: done

File: applications/asset_tracker_v2/doc/sensor_module.rst [hunk 26]
```diff
@@ -104,15 +104,16 @@ Bosch Software Environmental Cluster (BSEC) library
 The sensor module supports integration with the BSEC signal processing library using the external sensors, internal convenience API.
 If enabled, the BSEC library is used instead of the BME680 Zephyr driver to provide sensor readings from the BME680 for temperature, humidity, and atmospheric pressure.
 In addition, the BSEC driver provides an additional sensor reading, indoor air quality (IAQ), which is a metric given in between 0-500 range, which estimates the air quality of the environment.
+In the beginning, the IAQ will show as 50 (good air), but is automatically calibrated over time.
```
Comment (divipillai) [inline applications/asset_tracker_v2/doc/sensor_module.rst:107]:
```suggestion
In the beginning, the IAQ shows 50 (good air), but it is automatically calibrated over time.
```

File: doc/nrf/drivers/bme68x_iaq.rst [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
+#################
+
+.. contents::
+   :local:
+   :depth: 2
+
+You can use the BME68X IAQ driver to run the Bosch Sensor Environmental Cluster (BSEC) library in order to get Indoor Air Quality (IAQ) readings.
+
+.. note::
+   Using the BSEC library requires accepting a separate license agreement.
```
Comment (bjorn-tore-taraldsen) [inline doc/nrf/drivers/bme68x_iaq.rst:13]: Want to be even more precise on that this is a different license and needs to be handled/ downloaded by customer.
Suggested text: 
The BSEC library is distributed with a Bosch proprietary [license](https://www.bosch-sensortec.com/media/boschsensortec/downloads/software/bme688_development_software/2023_04/license_terms_bme688_bme680_bsec.pdf). that blocks it from been distributed as part of our nRF Connect SDK. To start using it you have to accept license and enable the download as described.

File: doc/nrf/drivers/bme68x_iaq.rst [hunk 28]
```diff
@@ -0,0 +1,39 @@
...
+.. contents::
+   :local:
+   :depth: 2
+
+You can use the BME68X IAQ driver to run the Bosch Sensor Environmental Cluster (BSEC) library in order to get Indoor Air Quality (IAQ) readings.
+
+.. note::
+   The BSEC library is distributed with a Bosch proprietary license (`BSEC license`_) that blocks it from been distributed as part of our nRF Connect SDK.
+
+To start using it you have to accept license and enable the download with the following commands:
```
Comment (gregersrygg) [inline doc/nrf/drivers/bme68x_iaq.rst:15]:
```suggestion
To start using it you have to accept the license and enable the download with the following commands:
```

**Review by thst-nordic — APPROVED** [review]
looks good.
the two added repos are ~4mb so won't affect 'west update' time

File: doc/nrf/drivers/bme68x_iaq.rst [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+
+.. note::
+   The BSEC library is distributed with a Bosch proprietary license (`BSEC license`_) that blocks it from been distributed as part of our nRF Connect SDK.
+
+To start using it you have to accept the license and enable the download with the following commands:
+
+.. code-block::
+
+   west config manifest.group-filter +bsec
+   west update
```
Comment (divipillai) [inline doc/nrf/drivers/bme68x_iaq.rst:20]: The note could be removed in this case because line no: 15 is a continuation of line no:13.
```suggestion
The BSEC library is distributed with a Bosch proprietary license (`BSEC license`_) that prevents it from being part of |NCS|.
To start using it, you have to accept the license and enable the download with the following commands:

.. code-block::

   west config manifest.group-filter +bsec
   west update
```

**Review by gregersrygg — APPROVED** [review]

File: west.yml [hunk 30]
```diff
@@ -251,6 +251,16 @@ manifest:
       revision: 384a091445c57b44ac8cbd18ebd245b47c71db94
       import:
         path-prefix: tools
+    - name: bme68x
+      url: https://github.com/boschsensortec/Bosch-BME68x-Library
```
Comment (carlescufi) [inline west.yml:255]: Please create a remote for this

File: west.yml [hunk 31]
```diff
@@ -251,6 +251,16 @@ manifest:
...
+      path: modules/lib/bme68x
+      revision: v1.1.40407
+    - name: bsec
+      url: https://github.com/boschsensortec/Bosch-BSEC2-Library
```
Comment (carlescufi) [inline west.yml:259]: Please use the previously created remote

File: west.yml [hunk 32]
```diff
@@ -251,6 +251,16 @@ manifest:
       revision: 384a091445c57b44ac8cbd18ebd245b47c71db94
       import:
         path-prefix: tools
+    - name: bme68x
+      url: https://github.com/boschsensortec/Bosch-BME68x-Library
+      path: modules/lib/bme68x
+      revision: v1.1.40407
```
Comment (carlescufi) [inline west.yml:257]: this one doesn´t need to be in the `bsec` group?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: west.yml (hunk 31 above)
Comment (maxd-nordic) [reply west.yml]: done

File: west.yml (hunk 30 above)
Comment (maxd-nordic) [reply west.yml]: done

**Review by carlescufi — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
